source/2018-05-02-api-evolution-for-rest-http-apis.html.markdown

```diff
@@ -0,0 +1,127 @@
+
+At some point change cannot be prevented, so at that time evolution suggests you provide sensible warnings to clients, letting them know if a feature they're using is going away, and not bothering them otherwise.
+
+## Examples
+
+> The field `name` exists, and that needs to be splint into `first_name` and `last_name`.
+
+Easy enough. However the data is handled internally (splitting on first space or last space or some other [falsehood defying assumption](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/)) you now have two new fields.
+
+The serializer can change from outputting just their name, to outputting all three fields:
+
+```
+class UserSerializer
+  include FastJsonapi::ObjectSerializer
+
+  attributes :name, :first_name, :last_name
+
+  attribute :name do |object|
+    "#{object.first_name} #{object.last_name}"
+  end
+end
+```
+
+When folks `POST` or `PATCH` to your API, if they send a `name` you can convert it, if they send `first_name` and `last_name` it'll get picked up fine on the serializer. Job done.
+
+> The field `price` needs to stop being dollars/pounds/whatever as we're starting to support currencies that don't fit into "unit" and "subunit".
+
+Ok! [Currencies are hard](https://gist.github.com/rgs/6509585), so this sort of thing is not uncommon. Let's start [using micros](https://support.siftscience.com/hc/en-us/articles/203869406-The-amount-field) instead.
+
+We don't want to just outright change this value from dollars to cents because then we'd start charging $1,000,000 for stuff that should only cost $1, and folks probably wouldn't like that. Let's make a new field: `price_micros`.
+
+Now clients can either send the `price` field, and it'll convert. This will only work for the older currencies, and if `currency` is a field in the resource (or something nearby) then it's easy enough to support `price` for whatever initial currencies you had (dollar/pound/euro) and throw an error if somebody tries using price for these newer currencies, pointing them instead to the micro field.
+
+Nothing broke for existing use cases, and new functionality was added seamlessly.
+
+> We have too many old fields kicking around, we need to get rid of them.
+
+Deprecations can be communicated in a few ways for API's. For those using OpenAPI v3, you can mark it as `deprecated: true` in the documentation. That's no ideal, of course, as OpenAPI is usually human-readable documentation, sat out of band on a developer portal somewhere. Rarely are OpenAPI schemas shoved into the response like JSON Schema is, so programatically clients have no real way to access this.
```

Comment:
typo: should be... `programmatically` (I think)

Reply:
Thanks, fixed!